stl/src/atomic.cpp

```diff
@@ -4,33 +4,87 @@
 // implement shared_ptr spin lock
 
 #include <yvals.h>
-
+#include <atomic>
 #include <intrin.h>
-#pragma warning(disable : 4793)
+#include <libloaderapi.h>
+#include <synchapi.h>
 
-_EXTERN_C
+#pragma warning(disable : 4793) // '%s' : function is compiled as native code
 
-// SPIN LOCK FOR shared_ptr ATOMIC OPERATIONS
-volatile long _Shared_ptr_flag;
+namespace {
+    // MUTEX FOR shared_ptr ATOMIC OPERATIONS
+    SRWLOCK _Shared_ptr_lock = SRWLOCK_INIT;
 
-_CRTIMP2_PURE void __cdecl _Lock_shared_ptr_spin_lock() { // spin until _Shared_ptr_flag successfully set
-#ifdef _M_ARM
-    while (_InterlockedExchange_acq(&_Shared_ptr_flag, 1)) {
-        __yield();
+#if _STL_WIN32_WINNT < _STL_WIN32_WINNT_VISTA

```

Comment:
This may become unnecessary due to SHA1 stuff. Let me follow up with setup folks.

Reply:
Setup folks confirmed today that this XP support machinery can be removed. (We still need to maintain ABI compatibility with static libs that were built with the `v140_xp` and `v141_xp` toolsets, but this change does that)

Reply:
Updated the change